### PR TITLE
add `htmlhelp_ascii_output`, `htmlhelp_binary_toc` and `htmlhelp_binary_index` options to htmlhelp builder

### PR DESCRIPTION
**Pull request by ghost**
### Feature or Bugfix
- Feature

### Purpose
- `htmlhelp_ascii_output` escapes non-ASCII characters in content's html file to ASCII.
- `htmlhelp_binary_toc` turns on Binary TOC in .hpp file.
- `htmlhelp_binary_index` turns on Binary Index in .hpp file.

### Detail
- `htmlhelp_ascii_output` fixes a problem that non-ASCII characters can not display in chm on some MBCS Windows systems, https://github.com/sphinx-doc/sphinx/issues/5788

### Relates
- https://github.com/sphinx-doc/sphinx/issues/5788

### CODE REVIEW

File: sphinx/builders/htmlhelp.py [hunk 1]
```diff
@@ -212,7 +212,10 @@ def open_file(self, outdir, basename, mode='w'):
 
     def update_page_context(self, pagename, templatename, ctx, event_arg):
         # type: (unicode, unicode, Dict, unicode) -> None
-        ctx['encoding'] = self.encoding
+        if self.config.htmlhelp_ascii_output:
+            ctx['encoding'] = 'ascii'
+        else:
+            ctx['encoding'] = self.encoding
```
Comment (tk0miya) [inline sphinx/builders/htmlhelp.py:218]: If my understanding is right, this is a hack to choose encoding via configuration. If true, I feel `htmlhelp_encoding` is much better for this case.

File: sphinx/builders/htmlhelp.py [hunk 2]
```diff
@@ -70,8 +70,8 @@
 
 project_template = '''\
 [OPTIONS]
-Binary TOC=No
-Binary Index=No
+Binary TOC=%(binary_toc)s
+Binary Index=%(binary_index)s
```
Comment (tk0miya) [inline sphinx/builders/htmlhelp.py:74]: What "Binary" means? Is there any description for these options?

Comment (ghost) [reply sphinx/builders/htmlhelp.py:74]: I write a long explanation, please read https://github.com/sphinx-doc/sphinx/issues/5788

**Comment by ghost** [comment]
A new PR to replace this one: https://github.com/sphinx-doc/sphinx/pull/5863

**Pull request closed without merging**